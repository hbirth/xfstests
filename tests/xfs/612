#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2021 Oracle.  All Rights Reserved.
#
# FS QA Test No. 612
# 
# Check that we can upgrade v5 only features on a v4 file system

. ./common/preamble
_begin_fstest auto quick

# Import common functions.
. ./common/filter

# real QA test starts here
_supported_fs xfs
_require_scratch_xfs_inobtcount
_require_command "$XFS_ADMIN_PROG" "xfs_admin"
_require_xfs_repair_upgrade inobtcount

# Make sure we can't upgrade to inobt on a V4 filesystem
_scratch_mkfs -m crc=0,inobtcount=0,finobt=0 >> $seqres.full
_scratch_xfs_admin -O inobtcount=1 2>> $seqres.full
_check_scratch_xfs_features INOBTCNT

# Make sure we can't upgrade to bigtim on a V4 filesystem
_scratch_mkfs -m crc=0 >> $seqres.full
_scratch_xfs_admin -O bigtime=1 2>> $seqres.full
_check_scratch_xfs_features BIGTIME

status=0
exit
