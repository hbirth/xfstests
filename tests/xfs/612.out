QA output created by 612
Running xfs_repair to upgrade filesystem.
Inode btree count feature only supported on V5 filesystems.
FEATURES: INOBTCNT:NO
Running xfs_repair to upgrade filesystem.
Large timestamp feature only supported on V5 filesystems.
FEATURES: BIGTIME:NO
